buildSrc/src/main/java/net/neoforged/neodev/NeoDevPlugin.java

```diff
@@ -528,4 +550,66 @@ static TaskProvider<CreateMinecraftArtifacts> configureMinecraftDecompilation(Pr
             task.getNeoFormArtifact().set(mcAndNeoFormVersion.map(version -> "net.neoforged:neoform:" + version + "@zip"));
         });
     }
+
+    private void setupProductionClientTest(Project project,
+                                      NeoDevConfigurations configurations,
+                                      TaskProvider<? extends DownloadAssets> downloadAssets,
+                                      TaskProvider<? extends AbstractArchiveTask> installer,
+                                      Provider<String> minecraftVersion,
+                                           Provider<String> neoForgeVersion,
+                                           Provider<RegularFile> originalClientJar
+    ) {
+
+        var installClient = project.getTasks().register("installProductionClient", InstallProductionClient.class, task -> {
+            task.setGroup(INTERNAL_GROUP);
+            task.setDescription("Runs the installer produced by this build and installs a production client.");
+            task.getInstaller().from(installer.flatMap(AbstractArchiveTask::getArchiveFile));
+
+            var destinationDir = project.getLayout().getBuildDirectory().dir("production-client");
+            task.getInstallationDir().set(destinationDir);
+        });
+
+        Consumer<RunProductionClient> configureRunProductionClient = task -> {
+            task.getLibraryFiles().addAll(IdentifiedFile.listFromConfiguration(project, configurations.neoFormClasspath));
+            task.getLibraryFiles().addAll(IdentifiedFile.listFromConfiguration(project, configurations.launcherProfileClasspath));
+            task.getAssetPropertiesFile().set(downloadAssets.flatMap(DownloadAssets::getAssetPropertiesFile));
+            task.getMinecraftVersion().set(minecraftVersion);
+            task.getNeoForgeVersion().set(neoForgeVersion);
+            task.getInstallationDir().set(installClient.flatMap(InstallProductionClient::getInstallationDir));
```

Comment:
Note that this creates a dependency on the `installClient` task. Is this intended?

Reply:
Just checking. :grin: